environment/src/test/java/com/sequenceiq/environment/environment/encryption/EnvironmentEncryptionServiceTest.java

```diff
@@ -133,4 +147,88 @@ void testShouldThrowExceptionWhenCloudPlatformNotAzure() {
+                .findFirst();
+        assertNotNull(dummyResource);
+        assertEquals(dummyResource.get().getReference(), "dummyDesId");
+    }
+
+    @Test
+    void testDeleteEncryptionResourcesShouldCallDeleteDiskEncryptionSetWhenCloudPlatformAzure() {
+        when(cloudPlatformConnectors.get(any(CloudPlatformVariant.class))).thenReturn(cloudConnector);
+        when(cloudConnector.encryptionResources()).thenReturn(encryptionResources);
+        EnvironmentDto environmentDto = EnvironmentDto.builder()
+                .withResourceCrn("crn:cdp:environments:us-west-1:dummyUser:environment:randomGeneratedResource")
+                .withId(1L)
+                .withName("envName")
+                .withCloudPlatform("AZURE")
+                .withCredential(new Credential())
+                .withLocationDto(LocationDto.builder()
+                        .withName("dummyRegion")
+                        .build())
+                .withParameters(ParametersDto.builder()
+                        .withAzureParameters(AzureParametersDto.builder()
+                                .withEncryptionParameters(AzureResourceEncryptionParametersDto.builder()
+                                        .withDiskEncryptionSetId("dummyDesId").build())
+                                .build())
+                        .build())
+                .withCreator("dummyUser")
+                .withAccountId("dummyAccountId")
+                .withCredential(new Credential())
+                .build();
+        CloudResource desCloudResource = CloudResource.builder()
+                .name("Des")
+                .type(ResourceType.AZURE_DISK_ENCRYPTION_SET)
+                .reference("dummyDesId")
+                .status(CommonStatus.CREATED)
+                .build();
+        when(resourceRetriever.findByResourceReferenceAndStatusAndType(any(), any(), any())).thenReturn(Optional.ofNullable(desCloudResource));
+        underTest.deleteEncryptionResources(environmentDto);
+        verify(encryptionResources).deleteDiskEncryptionSet(diskEncryptionSetDeletionRequestCaptor.capture());
+        verifyDiskEncryptionSetDeletionRequest();
+    }
+
+    private void verifyDiskEncryptionSetDeletionRequest() {
+        DiskEncryptionSetDeletionRequest deletionRequest = diskEncryptionSetDeletionRequestCaptor.getValue();
+        CloudContext cloudContext = deletionRequest.getCloudContext();
+        List<CloudResource> cloudResources = deletionRequest.getCloudResources();
+        Optional<CloudResource> desCloudResourceOptional = cloudResources.stream()
+                .filter(r -> r.getType() == ResourceType.AZURE_DISK_ENCRYPTION_SET)
+                .findFirst();
+        assertEquals(desCloudResourceOptional.get().getReference(), "dummyDesId");
+        assertEquals(desCloudResourceOptional.get().getName(), "Des");
+        assertEquals(desCloudResourceOptional.get().getType(), ResourceType.AZURE_DISK_ENCRYPTION_SET);
```

Comment:
Nit:

- `desCloudResourceOptional.get()` could be extracted to reduce repetitions.
- Before these checks it might be worth asserting something like `assertTrue(desCloudResourceOptional.isPresent())`. Without this, the first invocation of `desCloudResourceOptional.get()` will fail with `NoSuchElementException` if there is no `AZURE_DISK_ENCRYPTION_SET` resource found in `cloudResources`. The test case will surely abort in any case, so this is more of a personal preference to make the `isPresent()` check first. 😄